cloudinit/config/cc_ubuntu_advantage.py

```diff
@@ -247,7 +280,64 @@ def maybe_install_ua_tools(cloud):
         raise
 
 
-def handle(name, cfg, cloud, log, args):
+def _is_pro() -> bool:
+    try:
+        from uaclient.api.u.pro.attach.auto.should_auto_attach.v1 import (
+            should_auto_attach,
+        )
+    except ImportError as ex:
+        LOG.debug("Unable to import `uaclient`: %s", ex)
+        return False
+    try:
+        result = should_auto_attach()
+    except Exception as ex:
+        LOG.debug("Error during `should_auto_attach`: %s", ex)
+        LOG.warning(
+            "Unable to determine if this is an Ubuntu Pro instance."
+            " Fallback to normal UA attach."
+        )
+        return False
+    return result.should_auto_attach
+
+
+def _attach(ua_section: dict, config: Optional[dict] = None):
+    token = ua_section.get("token")
+    if not token:
+        msg = "`ubuntu-advantage.token` required in non-Pro Ubuntu instances."
+        LOG.error(msg)
+        raise RuntimeError(msg)
+    configure_ua(
+        token=token,
+        enable=ua_section.get("enable"),
+        config=config,
+    )
+
+
+def _auto_attach(ua_section: dict):
+    try:
+        from uaclient.api.u.pro.attach.auto.full_auto_attach.v1 import (
+            FullAutoAttachOptions,
+            full_auto_attach,
+        )
+        from uaclient.exceptions import UserFacingError
```

Comment:
Just a recommendation, you could import it from `uaclient.api.exceptions`; just as stated before, it is so that we maintain a list of exceptions for the api endpoints.

Reply:
Done.